src/ServiceControl.Monitoring/Timings/LongValueOccurrenceHandler.cs

```diff
@@ -0,0 +1,40 @@
+﻿namespace ServiceControl.Monitoring.Timings
+{
+    using System;
+    using System.Threading.Tasks;
+    using NServiceBus;
+    using Processing.RawData.NServiceBus.Metrics;
+    using Processing.Snapshot;
+
+    class TimingsReportHandler : IHandleMessages<LongValueOccurrences>
+    {
+        readonly TimingsStore processingTimeStore;
+        readonly CriticalTimeStore criticalTimeStore;
+
+        static string ProcessingTimeMessageType = "NServiceBus.Metrics.ProcessingTime";
+        static string CriticalTimeMessageType = "NServiceBus.Metrics.CriticalTime";
+
+        public TimingsReportHandler(ProcessingTimeStore processingTimeStore, CriticalTimeStore criticalTimeStore)
+        {
+            this.processingTimeStore = processingTimeStore;
+            this.criticalTimeStore = criticalTimeStore;
+        }
+
+        public Task Handle(LongValueOccurrences message, IMessageHandlerContext context)
+        {
+            var messageType = context.MessageHeaders[Headers.EnclosedMessageTypes];
```

Comment:
`EnclosedMessageTypes` is already an NServiceBus header and we're kind of using it differently. Would it be OK to call this `MetricTypes`?